### PR TITLE
807 step by step

### LINKED ISSUE #675: Add Heroku example to development.md
When  referencing Heroku, one could point out/give example how to do that with heroku, i.e. adding an example with a Config to load demo project as heroku app

### LINKED ISSUE #677: change wording in /docs/source/index.rst
> **Multiple configuration sources** While Nivio has its proprietary format,...

It is a proprietary format but I would also stress it is a yaml format. Proprietary sound more like binary format...

### LINKED ISSUE #679: Where to put petclinic demo (/docs/source/assessment.rst?)
Comments from Daniel L.:

Would it make sense to use the petclinic example (custom KPI mrem definition) in the assessment.rst file?
or
Do you want to describe/explain the petclinic example in an appendix and maybe link the corresponding directives?

### LINKED ISSUE #682: API documentation missing
Comments from Daniel L.:

Is there an API documentation?

### LINKED ISSUE #807: Step by Step Guide
As a new user who wants to set up nivio for the first time I need a simple step by step guide to get nivio running with my data.

## Acceptance criteria

- [ ] use contents from https://nebulous-talon-680.notion.site/Getting-started-with-Nivio-b14bc3ea971a4b79818375a73cb5f089

### LINKED ISSUE #810: Many small groups with lots of relations still collide
In order to mitigate layout problems reintroduce layout config options.

## Acceptance criteria

- [ ] min/max distances for group and item layouts be be configured#
- [ ] initial temps can be configured
- [ ] documentation is updated

### LINKED ISSUE #812: Weird closing behaviour after opening statuses
**Describe the bug**
Weird closing behaviour after opening statuses, too many closing icons appear

**To Reproduce**
1. open Status side bar
2. click Warnings gehen
3.  click an item in the list
4. multiple closing icons appear

**Expected behavior**
Only one should appear and close the sidebar.

**Screenshots**

image

image

**Additional context**
Leftover from sidebar refactoring

### LINKED ISSUE #813: KPIs etc not updated properly on push
**Describe the bug**
When new KPI config is pushed to the HTTP API the landscape config is not updated properly.

**To Reproduce**
* disable one default KPI, e.g. scaling ... disables all KPIs
* removing the config completely does not update

**Expected behavior**
Config is updated completely, not partially

### LINKED ISSUE #815: Mitigate CVE-2021-44228
Prepare against https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228. 

This is not a major concern in nivio as in other projects, since we're runnning Java 17 and do not use such logging statements. However, the possibility of being vulnerable to this exploit should be mitigated.

Proposed fix in pom:

```xml
  <properties>
          <!-- ... -->
          <log4j2.version>2.15.0</log4j2.version>
    </properties>
```

See also https://app.snyk.io/org/bonndan/project/57bbd428-b835-46d9-9362-8a5d20323947#issue-SNYK-JAVA-ORGAPACHELOGGINGLOG4J-2314719

## Acceptance criteria

- [ ] mvn dependency:tree shows needed log4j version

### LINKED ISSUE #819: Group attributes not updated on PUSH
Nivio seems to struggle with updating group attributes; once the attributes like `owner` or `description` have been assigned, you need to restart Nivio and then submit the landscape again with the new values.

### PR DESCRIPTION
**Pull request by bonndan**
closes #675
closes #677
closes #679
closes #682 
closes #807
closes #810 
closes #812
closes #813 
closes #815 
closes #819

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/model/Process.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    public List<? extends Assessable> getChildren() {
+        return null;
+    }
+
+    @Override
+    public String getLabel(String key) {
+        return null;
```
Comment (robrobert99) [inline src/main/java/de/bonndan/nivio/model/Process.java:33]: Returning null is not the best practice.

File: src/main/java/de/bonndan/nivio/model/Process.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    public String getAssessmentIdentifier() {
+        return null;
+    }
+
+    @Override
+    public List<? extends Assessable> getChildren() {
+        return null;
```
Comment (robrobert99) [inline src/main/java/de/bonndan/nivio/model/Process.java:28]: Returning null is not the best practice.

File: src/main/java/de/bonndan/nivio/model/Process.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private final Map<String, String> labels = new HashMap<>();
+
+    @Override
+    public Set<StatusValue> getAdditionalStatusValues() {
+        return null;
+    }
+
+    @Override
+    public String getAssessmentIdentifier() {
+        return null;
```
Comment (robrobert99) [inline src/main/java/de/bonndan/nivio/model/Process.java:23]: Returning null is not the best practice.

File: src/main/java/de/bonndan/nivio/model/Process.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Set;
+
+public class Process implements Linked, Tagged, Labeled, Assessable {
+
+    private final Map<String, Link> links = new HashMap<>();
+    private final Map<String, String> labels = new HashMap<>();
+
+    @Override
+    public Set<StatusValue> getAdditionalStatusValues() {
+        return null;
```
Comment (robrobert99) [inline src/main/java/de/bonndan/nivio/model/Process.java:18]: Returning null is not the best practice.

File: src/main/java/de/bonndan/nivio/model/Process.java (hunk 1 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/model/Process.java:33]: oh sh*t, I accidentally committed this file. 🙈

File: readme.md [hunk 5]
```diff
@@ -28,14 +28,13 @@ You need [docker](https://docker.com) to run the demo:
 
     docker run -e DEMO=1 -p 8080:8080 dedica/nivio
 
-Point your browser to http://localhost:8080/#/landscape/petclinic
+Point your browser to http://localhost:8080/#/landscape/petclinic to explore the features. An OpenAPI documentation is available under http://localhost:8080/swagger-ui.html.
```
Comment (jenarp) [inline readme.md:32]: An OpenAPI documentation is available at http://localhost:8080/swagger-ui.html.

Comment (bonndan) [reply readme.md:32]: fixed

**Review by jenarp — APPROVED** [review]

**Pull request merged**